x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java

```diff
@@ -127,15 +124,9 @@ public void testFullClusterRestart() throws Exception {
     public void testCloseUnassignedJobAndDatafeed() throws Exception {
         internalCluster().ensureAtMostNumDataNodes(0);
         logger.info("Starting dedicated master node...");
-        internalCluster().startNode(Settings.builder()
-                .put("node.master", true)
-                .put("node.data", true)
-                .put("node.ml", false)
-                .build());
+        internalCluster().startMasterOnlyNode();
         logger.info("Starting ml and data node...");
-        internalCluster().startNode(Settings.builder()
-                .put("node.master", false)
-                .build());
+        internalCluster().startNode(onlyRoles(Set.of(DiscoveryNodeRole.DATA_ROLE, DiscoveryNodeRole.MASTER_ROLE)));
```

Comment:
The code here used to disable `master`, but now it is enabled. Which is correct?